src/coreclr/nativeaot/Runtime/amd64/ExceptionHandling.asm

```diff
@@ -534,20 +534,23 @@ endif
         ;; if have shadow stack, then we need to reconcile it with the rsp change we have just made
         rdsspq  r9
         test    r9, r9
-        jz      NoSSP
+        jz      SSP_OK
 
         ;; Find the shadow stack pointer for the frame we are going to restore to.
         ;; The SSP we search is pointing to the return address of the frame represented
         ;; by the passed in context. So we search for the instruction pointer from 
-        ;; the context and return one slot up from there.
-        ;; (Same logic as in GetSSPForFrameOnCurrentStack)
+        ;; the context or for RhpThrowHwEx2, which is there in a case of a hardware fault,
+        ;; and pop SSP to one slot up from there.
+        mov     r8, RhpThrowHwEx2
         xor     r11, r11
    @@:  inc     r11
+        cmp     [r9 + r11 * 8 - 8], r8
+        je      SSP_OK
```

Comment:
This change doesn't modify the SSP when it finds the RhpThrowHwEx2. It seems you wanted to jump to the incsspq location?

Reply:
Ah, right.

Reply:
The PR was missing a commit.

Reply:
Also the IP that we look for normally is not there only when the AV and catch are in the same method. 
(that is why only a single test is affected)

If we see the IP on shadow stack, we should prefer that. Otherwise look for RhpThrowHwEx2.